tests/integration/suite/test_rbac.py

```diff
@@ -457,19 +466,35 @@ def test_member_can_perform_app_action(admin_mc, admin_pc, remove_resource,
     wait_until_available(user.client, project)
 
     app = client.create_app(
-        name="testapp1",
-        externalId="catalog://?catalog=library&template=mysql&version=0.3.7&"
+        name="test-" + random_str(),
+        externalId="catalog://?catalog=library&template"
+                   "=mysql&version=0.3.7&"
                    "namespace=cattle-global-data",
         targetNamespace=ns.name,
         projectId=project.id
     )
 
-    wait_for_workload(client, ns.name, count=1)
+    # if upgrade is performed prior to installing state,
+    # it may return a modified error
+    def is_installing():
+        current_state = client.reload(app)
+        if current_state.state == "installing":
+            return True
+        return False
+
+    try:
+        wait_for(is_installing)
+    except Exception as e:
+        # a timeout here is okay, the intention of the wait_for is to reach a
+        # steady state, this test is not concerned with whether an app reaches
+        # installing state or not
+        assert "Timeout waiting for condition" in str(e)
 
     user.client.action(
         obj=app,
         action_name="upgrade",
         answers={"asdf": "asdf"})
+
```

Comment:
Should we be asserting anything here about the response from `user.client.action` like a 200?